akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

```diff
@@ -249,14 +269,35 @@ private[cluster] case class HeartbeatNodeRing(selfAddress: UniqueAddress, nodes:
   /**
    * The receivers to use from a specified sender.
    */
-  def receivers(sender: UniqueAddress): immutable.Set[UniqueAddress] =
+  def receivers(sender: UniqueAddress): Set[UniqueAddress] =
     if (useAllAsReceivers)
       nodeRing - sender
     else {
-      val slice = nodeRing.from(sender).tail.take(monitoredByNrOfMembers)
-      if (slice.size < monitoredByNrOfMembers)
-        (slice ++ nodeRing.take(monitoredByNrOfMembers - slice.size))
-      else slice
+
+      // Pick nodes from the iterator until n nodes that are not unreachable have been selected.
+      // Intermediate unreachable nodes are also included in the result.
```

Comment:
The primary scenario that this solves is if all other nodes but one are shutdown.

> nodes that has been shut down together might all actually monitoring each other forming a clique

I don't think that can happen. At least it should be very unlikely. Each node watch 5 other nodes, and they are **not** picked randomly.

> if this node has monitored none of the shutdown ones, then no extension happens

If you have 10 nodes and shutdown 7 of them, there will be 3 left, and someone of these will have monitored the shutdown node. There could probably be some case where it is not covered for a series of shutdowns in chunks, before the information is spread.